runtime/src/main/java/org/corfudb/runtime/view/StreamsView.java

```diff
@@ -131,91 +131,75 @@ public void gc(long trimMark) {
     public long append(@Nonnull Object object, @Nullable TxResolutionInfo conflictInfo,
                        @Nonnull CacheOption cacheOption, @Nonnull UUID... streamIDs) {
 
+        final boolean serializeMetadata = false;
         final LogData ld = new LogData(DataType.DATA, object, runtime.getParameters().getCodecType());
+        TokenResponse tokenResponse = null;
 
-        ld.checkMaxWriteSize(runtime.getParameters().getMaxWriteSize());
+        // Opening serialization handle before acquiring token, this way we prevent the
+        // readers to wait for the possibly long serialization time in writer.
+        // The serialization here only serializes the payload because the token is not
+        // acquired yet, thus metadata is incomplete. Once a token is acquired, the
+        // writer will append the serialized metadata to the buffer.
+        try (ILogData.SerializationHandle sh = ld.getSerializedForm(serializeMetadata)) {
+            int payloadSize = ld.checkMaxWriteSize(runtime.getParameters().getMaxWriteSize());
```

Comment:
This doesn't cover the case of writes are issued directly to the `AddressSpaceView::write` method (i.e. the checkpointer writer).

Reply:
It seems like the CheckpointWriter will use this path too.

Reply:
Any  request directly from AddressSpaceView::write will be serialized only once in ChainReplicationProtocol::write, that path should be fine, right?